src/Text/Layout/Table/Cell.hs

```diff
@@ -29,6 +29,34 @@ class Cell a where
     dropBoth :: Int -> Int -> a -> a
     dropBoth l r = dropRight r . dropLeft l
 
+    -- | Like 'dropLeft', but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of padding needed is returned with the
+    -- result.
+    dropLeftNoPad :: Int -> a -> (Int, a)
+    dropLeftNoPad n a = (0, dropLeft n a)
+
+    -- | Like 'dropRight, but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of padding needed is returned with the
+    -- result.
+    dropRightNoPad :: Int -> a -> (a, Int)
+    dropRightNoPad n a = (dropRight n a, 0)
+
+    -- | Like 'dropBoth, but does not add extra padding if the exact amount
+    -- can't be dropped. The amount of left and right padding needed is
+    -- returned with the result.
+    -- The default implementation, when invoked with @dropBothNoPad l r@, will
+    -- first drop @l@ from the left and then will drop an amount from the right
+    -- so that the total dropped is @l + r@. Note that this means it may drop
+    -- less than @r@ from the right if it is forced to drop more than @l@ from
+    -- the left.
+    dropBothNoPad :: Int -> Int -> a -> (Int, a, Int)
+    dropBothNoPad l' r' = go (max 0 l') (max 0 r')
```

Comment:
Why are the arguments called _l prime_ and _r prime_?  Usually this is used to express a (conceptual) _successor_ to a variable.